pyro/distributions/triangular.py

```diff
@@ -0,0 +1,77 @@
+from numbers import Number
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.utils import broadcast_all
+
+from pyro.distributions import TorchDistribution, Uniform
+
+
+class Triangular(TorchDistribution):
+    """
+     A Triangular distribution.
+
+    :param torch.Tensor low: lower range (inclusive).
+    :param torch.Tensor high: upper range (inclusive).
+    :param torch.Tensor peak: mode of range.
+    """
+    arg_constraints = {'low': constraints.real, 'high': constraints.dependent, 'peak': constraints.dependent}
+    support = constraints.dependent
+    has_rsample = False
+
+    @property
+    def mean(self):
+        return (self.low + self.high + self.peak) / 3
+
+    @property
+    def variance(self):
+        return (self.low**2 + self.high**2 + self.peak**2 - self.low * self.high - self.low * self.peak -
+                self.high * self.peak) / 18
+
+    def __init__(self, low, high, peak, validate_args=None):
+        self.low, self.high, self.peak = broadcast_all(low, high, peak)
+        self._uniform = Uniform(0, 1)
+
+        if isinstance(low, Number) and isinstance(high, Number) and isinstance(peak, Number):
+            batch_shape = torch.Size()
+        else:
+            batch_shape = self.low.shape
+        event_shape = torch.Size()
+        super(Triangular, self).__init__(batch_shape, event_shape, validate_args)
+
+        if self._validate_args:
+            if not torch.lt(self.low, self.high).all():
+                raise ValueError("Triangle is not defined when low >= high")
+            if not torch.le(self.low, self.peak).all():
+                raise ValueError("Triangle is not defined when peak < low")
+            if not torch.le(self.peak, self.high).all():
+                raise ValueError("Triangle is not defined when peak > high")
+
+    def sample(self, sample_shape=torch.Size()):
```

Comment:
I believe you can rename this to `rsample` since it is already correctly reparametrized.